### PR TITLE
Splitloren hints with prefix

### PR DESCRIPTION
**Pull request by newville**
addresses #566, this time for sure!

basic problem: parameter hints were not using prefix

###### Type of Changes

- [x] Bug fix

I did add a test this time!

### CODE REVIEW

**Comment by dimapu** [comment]
Dear Matt, 

Just noticed that in `test_height_and_fwhm_expression_evalution_in_builtin_models()`, line 130, the parameter name is `sigma_r`, while in `test_splitlorentzian_prefix()` the parameter name is `sigmar`. This looks strange to me.

**Comment by newville** [comment]
@dimapu Thanks -- I fixed the test to use `sigma_r`.

File: tests/test_lineshapes_models.py [hunk 1]
```diff
@@ -119,77 +119,87 @@ def test_finite_output_lineshape(lineshape):
...
         params.update_constraints()
+
+
+def test_splitlorentzian_prefix():
+    mod1 = models.SplitLorentzianModel()
+    par1 = mod1.make_params(amplitude=1.0, center=0.0, sigma=0.9, sigma_r=1.3)
+    par1.update_constraints()
+
+    mod2 = models.SplitLorentzianModel(prefix='prefix_')
+    par2 = mod2.make_params(amplitude=1.0, center=0.0, sigma=0.9, sigmai_r=1.3)
```
Comment (reneeotten) [inline tests/test_lineshapes_models.py]: almost correct, one more typo in this line ```sigmai_r``` ;)

**Comment by newville** [comment]
@reneeotten thanks, and yikes!  I would like to blame too much mixing of vim and emacs, but I think it is more likely that I'm overextended and not paying enough attention to each step at a time.  OK, good thing there are tests and people checking up on me.   This time for sure, and then "squash and merge"!

**Comment by mfdecamp** [comment]
Thanks.  One comment, in models.py the variable 'tiny' is not defined (Line 426 in models.py) which produced an error.  I manually substituted '1.e-13' for 'tiny' in line 426 to get the fitting routine to work.

**Comment by reneeotten** [comment]
> Thanks. One comment, in models.py the variable 'tiny' is not defined (Line 426 in models.py) which produced an error. I manually substituted '1.e-13' for 'tiny' in line 426 to get the fitting routine to work.

that statement is incorrect, it's defined right [here](https://github.com/lmfit/lmfit-py/blob/2ae2fc9e3dd6137d16cfa05474a34f33433e11ec/lmfit/models.py#L15). I would recommend either to stick with a released version or use the current ```master``` to get all recent changes. Otherwise you will need to make sure to manually apply the necessary changes from *all* commits yourself.

**Comment by newville** [comment]
@mfdecamp like @reneeotten says, it should be defined.  Can you post an example showing a problem?

**Comment by mfdecamp** [comment]
Thanks again.  That line was not included in the models.py file from the conda distribution as of 7/1/2019.  I manually inserted the definition and the fitting works.

**Comment by reneeotten** [comment]
Yes, that is correct - it cannot be present there because I added it in commit 9a10b462c6a831f422e5b5c596044f2306a90500, after version 0.9.13 was released. Selectively applying changes form only one commit isn't always guaranteed to work, but I am glad it's sorted out now and it will be all available in the next release!

**Pull request merged**